### DISCUSSION TITLE
Can't see database functions.

### DISCUSSION THREAD

**Discussion by KBearMan** [simolus3/drift#1877] [Q&A]
Has anyone else run into issues seeing the functions written in the database class? I can't seem to call them outside the database class. This has lead to me having to create the database and then reference it to make any sql calls. 

`
  Future<List<Profile>> getAllProfiles() async{
    return await _database.select(_database.profiles).get();
  }
`

instead of being able to put this function in the database class as 

`  Future<List<Profile>> get AllProfiles => select(_database.profiles).get();`

and calling

`_database.AllProfiles();`

**Comment by simolus3** [comment]
> seeing the functions written in the database class?

Do you mean things like `select`? These are defined on the database class itself, so they should only be visible on the database instance.

>  Future<List<Profile>> get AllProfiles => select(_database.profiles).get();

It would probably have to be `select(profiles).get()`, but what error do you get for that? This _should_ work.

**Reply by KBearMan** [reply]
No I mean if I define a function within the database class. In this example, AllProfiles.

```
class MyDatabase extends _$MyDatabase {
  MyDatabase() : super(_openConnection());

  @override
  int get schemaVersion => 1;

Future<List<Profile>> get AllProfiles => select(profiles).get();
}
```


And then elsewhere I've created the database instance.

`MyDatabase _database = MyDatabase();`


If I try to call AllProfiles from _database, I can't. The function is out of scope. If I define ANY function within the database class, I can't access it elsewhere. So now I'm having to define the select statements OUTSIDE of the database class. 
example: 
```

class myOtherClass{
MyDatabase _database = MyDatabase();

Future<List<Profile>> getAllProfiles() async{ return await _database.select(_database.profiles).get(); }

}
```

**Reply by simolus3** [reply]
That is pretty weird.

> The function is out of scope

Well it's a getter, so you'd just do `await _database.AllProfiles`, not `await _database.AllProfiles()`. But I guess that's not the problem here?

> If I try to call AllProfiles from _database, I can't. The function is out of scope.

I can try to help more if you post the exact message from the analyzer. Maybe it's something weird with the class, maybe it's an ambiguous reference, .... Hard to say for sure without seeing more details.

**Reply by KBearMan** [reply]
There were no analyzer complaints. Still not sure what went wrong but after a few tries of deleting the g file, running a flutter clean and re-running the build runner, it seems to have worked itself out?  I guess there was an issue with a cache.
